docs/environment.yml

```
@@ -5,7 +5,7 @@ name: rtd
 channels:
   - numba/label/dev
 dependencies:
-  - python=3.7
+  - python=3.8
```

Since this is for the docs build, we can take this straight to 3.10? Means we don't need to bump it every year.
```suggestion
  - python=3.10
```

---

Is there any chance of the docs build somehow ending up with "newer" python features in the source which would stop docs building on the oldest supported Python?

---

@esc Would you like this bumped to 3.10? Or just keep it set to the minimum supported version as done in this PR currently?

---

@jamesobutler I don't know yet, there doesn't seem to be any consensus on this yet. Both options appear to have legitimate pros and cons.

---

@esc Since this PR continues the current idea of using the minimum supported version of python for building documentation, can we proceed with this PR as is? Switching to a newer python version 3.10 or 3.11 can happen in a separate PR.

---

@jamesobutler that sounds like a reasonable resolution, thank you.